### DISCUSSION TITLE
Datepicker shows previous date

### DISCUSSION THREAD

**Discussion by courtneynrwilliams** [jrief/django-formset#174] [Q&A]
I'm trying to use the DatePicker. The current date shown in the datepicker is accurate. When I select a date, the date shown in the input box is the date before the date selected. I've also tested the DateTimePicker, and it does the same thing. DateInput, however, shows the correct selected date.

Why is this?
How do I fix this issue?

**Comment by jrief** [comment]
Hmm, I just tested with the date-picker as provided here: https://django-formset.fly.dev/date-time-input/#date-picker-widget
and everything seems fine. May I ask you in which timezone you're located and could this be the problem?

**Reply by courtneynrwilliams** [reply]
I'm in the US Central Time Zone

**Reply by courtneynrwilliams** [reply]
Here's what I have:

forms.py:

from django.forms import fields, forms
from formset.widgets import DatePicker, DateTimePicker, DateInput

class NewTaskForm (forms.Form):
    task = fields.CharField(label="My New Task", max_length=50)
    priority = fields.IntegerField(label="Priority", min_value=1, max_value=10)
    date = fields.DateField(label="Date", widget=DatePicker)



views.py:

from django.shortcuts import render
from formset.calendar import CalendarResponseMixin
from formset.views import FormView
from .forms import NewTaskForm

class NewTaskView (CalendarResponseMixin, FormView):
    form_class = NewTaskForm
    template_name = 'formtestapp/index.html'

# Create your views here.
def index(request):
    return render(request, 'formtestapp/index.html')



index.html:

<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8" name="viewport" content="initial-scale=1, width=device-width">
        <title>Hello</title>
        {% load static %}
        <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.3.3/dist/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-QWTKZyjpPEjISv5WaRU9OFeRpok6YctnYmDr5pNlyT2bRjXh0JMhjY6hW+ALEwIH" crossorigin="anonymous">
        <link href="{% static 'formset/css/bootstrap5-extra.css' %}" rel="stylesheet">
        <link href="{% static 'formset/css/collections.css' %}" rel="stylesheet">
        <script src="{% url 'javascript-catalog' %}"></script>
        <script type="module" src="{% static 'formset/js/django-formset.js' %}"></script>
    </head>
    <body>
        <div class="text-center">
            <h1>Hello World</h1>
        </div>
        {% load formsetify %}

        <div class="container">
            <django-formset endpoint="{{ request.path }}" csrf-token="{{ csrf_token }}">
                {% render_form form "bootstrap" field_classes="mb-2" %}
                <button type="button" df-click="submit -> proceed">Submit</button>
            </django-formset>
        </div>
    </body>
</html>



urls.py:

from django.urls import path
from django.views.i18n import JavaScriptCatalog
from formset.views import FormView

from . import views

urlpatterns = [
    path('jsi18n/', JavaScriptCatalog.as_view(packages=['formset']), name='javascript-catalog'),
    path('', views.NewTaskView.as_view(
        template_name='formtestapp/index.html',
    )),
]

**Comment by jrief** [comment]
Finally I found the reason why this is happening. Assuming your timezone is PST. In JavaScript, if you invoke `new Date('2025-01-18')` this returns "Fri Jan 17 2025 16:00:00 GMT-0800 (Pacific Standard Time)". On the other side, if you invoke `new Date('2025-01-18T00:00')` this returns "Sat Jan 18 2025 00:00:00 GMT-0800 (Pacific Standard Time)". This difference caused the error.

In version 1.6.1 this will be fixed. I had to refactor quite a bit of the code, so hopefully this didn't break other functionality.
